pkg/registry/apps/daemonset/storage/storage_test.go

```diff
@@ -193,4 +197,40 @@ func TestShortNames(t *testing.T) {
 	registrytest.AssertShortNames(t, storage, expected)
 }
 
-// TODO TestUpdateStatus
+func TestUpdateStatus(t *testing.T) {
```

Comment:
I think that it's not the right direction. We should finally do what we've done for all other tests.

If you look into say "TestUpdate" test, you see that it is generic and you just pass object-specific stuff to it.
And this is exactly the same we should do for UpdateStatus.

TestUpdateStatus in test should be pretty simple to TestUpdate, but instead of passing updateFunc, you should pass sth like "validStatus".

This will avoid duplicating the same code again and again.